test/src/test/groovy/org/openremote/test/protocol/http/HttpClientProtocolTest.groovy

```diff
@@ -616,9 +617,8 @@ class HttpClientProtocolTest extends Specification implements ManagerContainerTr
         String[] dynamicTimeTests = ["\${time:PT1H:yyyy-MM-dd HH:mm}", "\${time:PT2H:yyyy-MM-dd HH:mm:ss}", "\${time:P1D:yyyy-MM-dd}", "\${time:P2DT3H:yyyy-MM-dd HH:mm}", "\${time:PT30M:yyyy-MM-dd HH:mm}", "\${time:P7D:yyyy-MM-dd}", "\${time:PT5H:yyyy-MM-dd HH:mm}", "\${time:P2D:yyyy-MM-dd}", "\${time:PT1M:yyyy-MM-dd}", "\${time:P3D:yyyy-MM-dd}"] as List<String>
 
         then: "The strings are parsed correctly. If there is a way to use a constant timestamp, it'd be much easier."
-        TimeParameterInjectorFilterImpl filter = new TimeParameterInjectorFilterImpl(timerService::getCurrentTimeMillis);
         for (final String e in dynamicTimeTests){
-            def y = filter.replace(e);
+            def y = ProtocolUtil.doDynamicTimeReplace(e, timerService::getCurrentTimeMillis);
 
         }
```

Comment:
There is no assertion here and you mention constant timestamp, as you now know, the clock in tests can be stopped/started and set to any time